core/src/main/java/nl/nn/adapterframework/unmanaged/AbstractJmsConfigurator.java

```diff
@@ -34,32 +35,31 @@
  */
 abstract public class AbstractJmsConfigurator {
 	protected Logger log=LogUtil.getLogger(this);
-   
-    private IPortConnectedListener listener;
-    private ConnectionFactory connectionFactory;
-    private Destination destination;
+
+	private IPortConnectedListener<Message> listener;
+	private ConnectionFactory connectionFactory;
+	private Destination destination;
 	private ReceiverBase receiver;
-	private IbisExceptionListener exceptionListener;        
+	private IbisExceptionListener exceptionListener;
 
-    public void configureEndpointConnection(IPortConnectedListener listener, ConnectionFactory connectionFactory, Destination destination, IbisExceptionListener exceptionListener) throws ConfigurationException {
-    	if (connectionFactory==null) {
-    		throw new ConfigurationException("ConnectionFactory must be specified");
-    	}
-		if (destination==null) {
+	public void configureEndpointConnection(IPortConnectedListener<Message> listener, ConnectionFactory connectionFactory, Destination destination, IbisExceptionListener exceptionListener) throws ConfigurationException {
```

Comment:
In alle(?) gevallen geef je 'this' mee aan deze functie, kan je geen interface maken oid en dan de listener, credentails, exceptionlistener ed daar uit halen?

Reply:
Ja dat kan wel. Er kan natuurlijk van alles gerefactored worden, maar nu gingen we alleen zorgen dat er geauthenticeerde connecties gemaakt kunnen worden. 
Ik heb op dit moment geen JMS provider geinstalleerd, en wil er vooralsnog ook nog geen prioriteit aan geven. Jij wel?